app/api/server/v1/im.js

```diff
@@ -32,22 +32,53 @@ function findDirectMessageRoom(params, user) {
 
 API.v1.addRoute(['dm.create', 'im.create'], { authRequired: true }, {
 	post() {
-		const { username, usernames } = this.requestParams();
+		const { username, usernames, excludeSelf } = this.requestParams();
 
 		const users = username ? [username] : usernames && usernames.split(',').map((username) => username.trim());
+		const selfExcluded = excludeSelf && hasPermission(this.userId, 'view-room-administration');
 
 		if (!users) {
 			throw new Meteor.Error('error-room-not-found', 'The required "username" or "usernames" param provided does not match any direct message');
 		}
 
-		const room = Meteor.call('createDirectMessage', ...users);
+		const room = Meteor.call(excludeSelf ? 'createDirectMessageForOthers' : 'createDirectMessage', ...users);
+
+		// If we haven't included our self then we need to open the room for the other participants
+		if (selfExcluded) {
+			room.usernames.forEach((username) => {
+				const user = Users.findOneByUsernameIgnoringCase(username);
+				const subscription = Subscriptions.findOneByRoomIdAndUserId(room.rid, user._id);
+
+				if (!subscription.open) {
+					Meteor.runAsUser(user._id, () => {
+						Meteor.call('openRoom', room.rid);
+					});
+				}
+			});
+		}
```

Comment:
instead of doing all this, you can set `subscriptionExtra: { open: true }` when calling `createDirectMessage`

Reply:
Much nicer!